src/store/Slide/CartSlice.js

```diff
@@ -0,0 +1,38 @@
+import { createSlice } from '@reduxjs/toolkit'
+
+const cartItemLocalStorage = JSON.parse(localStorage.getItem('productItem'));
+
+const initialState = {
+    cartList: cartItemLocalStorage ? cartItemLocalStorage.cartList : [],
+    total :  cartItemLocalStorage ? cartItemLocalStorage.total : 0,
+    cartQuantity: cartItemLocalStorage ? cartItemLocalStorage.cartQuantity : 0,
+}
+
+export const CartSlice = createSlice({
+    name: 'cart',
+    initialState,
+    reducers: {
+        addToCart : (state, action) => {        
+            let existedItem = false;
+            state.total += action.payload.price
+            state.cartList.forEach((item) => {
+                if (item.id === action.payload.id) {
+                    item.quantity +=1;
+                    existedItem = true;
+                }
+            });
+
+			if (!existedItem) {
+				state.cartList.push(action.payload);
+				state.cartQuantity = state.cartList.length;
+			}
+
+			localStorage.setItem('productItem', JSON.stringify(state));
```

Comment:
Check lại indent, tìm hiểu thêm cách lưu data vào localstorage với redux
https://stackoverflow.com/questions/35305661/where-to-write-to-localstorage-in-a-redux-app
Với lại key của cái này phải là cart chứ sao lại là productItem nhỉ? productItem là 1 sản phẩm duy nhất thôi mà

Reply:
![image](https://user-images.githubusercontent.com/90613891/154403628-90656408-c5c6-41ba-9be1-562a021da62f.png)
em không biết sao trên vscode của em thì nó bình thường nhưng lên git lại bị thụt ạ